### PR TITLE
2023-01-09 - BryanSoltis - v2.4.0 Updates

### PR DESCRIPTION
**Pull request by BryanSoltis**

### CODE REVIEW

File: ready/AzNamingTool/Helpers/ConfigurationHelper.cs [hunk 1]
```diff
@@ -0,0 +1,467 @@
...
+                    // Set the result to cache
+                    CacheHelper.SetCacheObject("isconnected", result);
+                }
+                else
+                {
+                    result = (bool)data;
+                }
+            }
+            catch (Exception)
+            {
```
Comment (jocontr) [inline ready/AzNamingTool/Helpers/ConfigurationHelper.cs:182]: Please handle exception. You could have an array of them thrown in this routine.

File: ready/AzNamingTool/Helpers/ConfigurationHelper.cs [hunk 2]
```diff
@@ -0,0 +1,467 @@
...
+
+                    items = JsonSerializer.Deserialize<List<T>>(data, options);
+                }
+
+                return items;
+            }
+            catch (Exception ex)
+            {
+                AdminLogService.PostItem(new AdminLogMessage() { Title = "ERROR", Message = ex.Message });
+                return items;
```
Comment (jocontr) [inline ready/AzNamingTool/Helpers/ConfigurationHelper.cs:232]: You could have a single return path for easier future maintenance. https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/try-catch-finally

File: ready/AzNamingTool/Helpers/ConfigurationHelper.cs [hunk 3]
```diff
@@ -0,0 +1,467 @@
...
+                    {
+                        versiondata.Add("<h5>Resource Types</h5><hr /><div>Your resource types configuration is out of date!<br /><br />It is recommended that you refresh your resource types to the latest configuration.<br /><br /><strong>To Refresh:</strong><ul><li>Expand the <strong>Types</strong> section</li><li>Expand the <strong>Configuration</strong> section</li><li>Select the <strong>Refresh</strong> option</li></ul></div><br />");
+                    }
+
+                    // Resource Locations
+                    if (officialversiondata.resourcelocations != currentversiondata.resourcelocations)
+                    {
+                        versiondata.Add("<h5>Resource Locations</h5><hr /><div>Your resource locations configuration is out of date!<br /><br />It is recommended that you refresh your resource locations to the latest configuration.<br /><br /><strong>To Refresh:</strong><ul><li>Expand the <strong>Locations</strong> section</li><li>Expand the <strong>Configuration</strong> section</li><li>Select the <strong>Refresh</strong> option</li></ul></div><br />");
```
Comment (jocontr) [inline ready/AzNamingTool/Helpers/ConfigurationHelper.cs:388]: Not a fan of markup in code.

Is there a better strategy within this context?

File: ready/AzNamingTool/Helpers/ValidationHelper.cs [hunk 4]
```diff
@@ -228,5 +228,18 @@ public static bool ValidateShortName(string value, string type)
             }
         }
 
+        public static bool CheckNumeric(string value)
+        {
+            Regex regx = new("^[0-9]+$");
```
Comment (jocontr) [inline ready/AzNamingTool/Helpers/ValidationHelper.cs:233]: If you have more validations like this in the future you could generalize the two functions and create a rules dictionary.

**Review by jocontr — CHANGES_REQUESTED** [review]
Please address unhandled exception on static function checking for connectivity. That could give us a lot of headaches. Afterwards check in. Thank you

File: ready/AzNamingTool/Helpers/ConfigurationHelper.cs (hunk 1 above)
Comment (BryanSoltis) [reply ready/AzNamingTool/Helpers/ConfigurationHelper.cs:182]: This has been updated. The function now will log the exception to the Admin Log.

File: ready/AzNamingTool/Helpers/ConfigurationHelper.cs (hunk 3 above)
Comment (BryanSoltis) [reply ready/AzNamingTool/Helpers/ConfigurationHelper.cs:389]: Good point. I will research a cleaner/scalable way to implement this.

File: ready/AzNamingTool/Helpers/ConfigurationHelper.cs (hunk 2 above)
Comment (BryanSoltis) [reply ready/AzNamingTool/Helpers/ConfigurationHelper.cs]: Excellent suggestion. I updated the code to move the "return items" outside of the tr/catch.

**Review by jocontr — APPROVED** [review]

**Pull request merged**